src/marine/godae/ship2ioda.py

```diff
@@ -0,0 +1,171 @@
+
+        self.filename = filename
+        self.date = date
+
+        # Read ship data
+        self._rd_ship()
+
+        return
+
+    def _rd_ship(self):
+        '''
+        Read the surface obs data
+        Based on subroutine rd_ship in ocn_obs.f
+        '''
+
+        try:
+            fh = FortranFile(self.filename, mode='r', header_dtype='>i4')
+        except IOError:
+            raise IOError('%s file not found!' % self.filename)
+        except Exception:
+            raise Exception('Unknown error opening %s' % self.filename)
+
+        # odata is the dictionary with data structure as in ocn_obs.f
+        odata = {}
+
+        odata['n_obs'], odata['n_lvl'], odata['n_vrsn'] = fh.read_ints('>i4')
+
+        print('    number ship obs: %d' % odata['n_obs'])
+        print('  max number levels: %d' % odata['n_lvl'])
+        print('file version number: %d' % odata['n_vrsn'])
+
+        if odata['n_obs'] <= 0:
+            print('No ship observations to process from %s' % self.filename)
+            return
+
+        odata['ob_wm'] = fh.read_reals('>i4')
+        odata['ob_glb'] = fh.read_reals('>f4')
+        odata['ob_lat'] = fh.read_reals('>f4')
+        odata['ob_lon'] = fh.read_reals('>f4')
+        odata['ob_age'] = fh.read_reals('>f4')
+        odata['ob_clm'] = fh.read_reals('>f4')
+        odata['ob_qc'] = fh.read_reals('>f4')
+        odata['ob_rgn'] = fh.read_reals('>f4')
+        odata['ob_sst'] = fh.read_reals('>f4')
+        odata['ob_typ'] = fh.read_reals('>i4')
+        odata['ob_dtg'] = fh.read_record('>S12').astype('U12')
+        odata['ob_rcpt'] = fh.read_record('>S12').astype('U12')
+        odata['ob_scr'] = fh.read_record('>S1').astype('U1')
+
+        if odata['n_vrsn'] <= 2:
+            print('verify ob_sign for version = %d' % odata['n_vrsn'])
+            odata['ob_sign'] = fh.read_record('>S7').astype('U7')
+        else:
+            odata['ob_sign'] = fh.read_record('>S7').astype('U7')
+
+        if odata['n_vrsn'] > 1:
+            odata['ob_csgm'] = fh.read_reals('>f4')
+            odata['ob_gsgm'] = fh.read_reals('>f4')
+            odata['ob_rsgm'] = fh.read_reals('>f4')
+        else:
+            minus999 = np.ones(odata['n_obs'], dtype=np.float32) * -999.
+            odata['ob_csgm'] = minus999
+            odata['ob_gsgm'] = minus999
+            odata['ob_rsgm'] = minus999
+
+        fh.close()
+
+        self.odata = odata
+
+        return
+
+    def _to_nlocs(self):
+        '''
+        Selectively (e.g. salinity, temperature) convert odata into idata
+        idata is the IODA required data structure
+        '''
+
+        # idata is the dictionary containing IODA friendly data structure
+        idata = self.odata
+
+        # Diff. between ob. time and time in the file (hrs)!
+        for i in range(self.odata['n_obs']):
+            dtg = datetime.strptime(self.odata['ob_dtg'][i], '%Y%m%d%H%M')
+            idata['ob_dtg'][i] = (dtg - self.date).seconds/3600.
```

Comment:
the wrong values are being produced, this should use **.total_seconds()** instead

Reply:
same thing for profile2ioda.py as well

Reply:
...noticeable if you pass a date into the script that is different from the base date of the obs file

Reply:
indeed.